### PR TITLE
Adds temperature to solutions

### PR DESCRIPTION
**Pull request by TemporalOroboros**
Gives solutions a temperature and some basic backend to manipulate it.

**Changelog**

:cl:
- add: Added temperature handling to solutions. Nothing actually uses it yet, but it exists.

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
The tests are exploding because they run before `IPrototypeManager` is registered to `IoCManager`. How do I fix this?

File: Content.Shared/Chemistry/Components/Solution.cs [hunk 1]
```diff
@@ -94,24 +101,35 @@ public string GetPrimaryReagentId()
         /// </summary>
         /// <param name="reagentId">The prototype ID of the reagent to add.</param>
         /// <param name="quantity">The quantity in milli-units.</param>
-        public void AddReagent(string reagentId, FixedPoint2 quantity)
+        public void AddReagent(string reagentId, FixedPoint2 quantity, double? temperature = null)
         {
             if (quantity <= 0)
                 return;
+            if (!IoCManager.Resolve<IPrototypeManager>().TryIndex(reagentId, out ReagentPrototype? proto))
```
Comment (TemporalOroboros) [inline Content.Shared/Chemistry/Components/Solution.cs:108]: This line specifically is what is blowing up.

Comment (TemporalOroboros) [reply Content.Shared/Chemistry/Components/Solution.cs:108]: Paul has given me advice on how to fix this.

Comment (TemporalOroboros) [reply Content.Shared/Chemistry/Components/Solution.cs:108]: It looks like it's fixed.

Comment (TemporalOroboros) [reply Content.Shared/Chemistry/Components/Solution.cs:108]: It looks like it's fixed.

File: Content.Server/Chemistry/ReagentEffectConditions/SolutionThermalEnergy.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.Chemistry.ReagentEffectConditions
+{
+    /// <summary>
+    ///     Requires the solution to be above or below a certain temperature.
```
Comment (mirrorcult) [inline Content.Server/Chemistry/ReagentEffectConditions/SolutionThermalEnergy.cs:8]: incorrect xmldoc

File: Resources/Locale/en-US/administration/ui/manage-solutions/manage-solutions.ftl [hunk 3]
```diff
@@ -1,4 +1,9 @@
 admin-solutions-window-title = Solution Editor - {$targetName}
 admin-solutions-window-solution-label = Target solution:
 admin-solutions-window-add-new-button = Add new reagent
-admin-solutions-window-capacity-label = Capacity {$currentVolume}/{$maxVolume}u
+admin-solutions-window-volume-label = Volume {$currentVolume}/{$maxVolume}u
+admin-solutions-window-capacity-label = Capacity (u):
+admin-solutions-window-specific-heat-label = Specific Heat: {$specificHeat} J/(K*u)
+admin-solutions-window-heat-capacity-label = Heat Capacity: {$heatCapacity} J/K
+admin-solutions-window-temperature-label = Temperature (K):
+admin-solutions-window-thermal-energy-label = Thermal Energy (J):
```
Comment (mirrorcult) [inline Resources/Locale/en-US/administration/ui/manage-solutions/manage-solutions.ftl:9]: might want to look into the generic unit formatting in localization for some of the SI units here, but not strictly necessary

**Review by mirrorcult — CHANGES_REQUESTED** [review]
fantastic PR

having them be doubles is meh, I'd prefer if they were the same as atmos' temperature for consistency (float)

File: Content.Server/Chemistry/ReagentEffectConditions/SolutionThermalEnergy.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Server/Chemistry/ReagentEffectConditions/SolutionThermalEnergy.cs]: It has been corrected.

**Comment by TemporalOroboros** [comment]
They've been converted back to floats.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**